### DISCUSSION TITLE
Export the torch model of EfficientAD

### DISCUSSION THREAD

**Discussion by wwt123-star** [open-edge-platform/anomalib#1337] [Q&A]
I have no idea why the export  torch.pt of efficientad cannot work on inference?
The commond I used is as followed:
    inferencer = TorchInferencer(path='model.pt')

**Comment by blaz-r** [comment]
Can you provide more details please.

**Comment by samet-akcay** [comment]
@wwt123-star, can you use the issue template to provide more details?

**Comment by wwt123-star** [comment]
When I run the demo, this error occurs:
![1](https://github.com/openvinotoolkit/anomalib/assets/67778649/c63dcb98-34e6-4947-b6af-8a16a3da7e0d)

**Reply by blaz-r** [reply]
Hello. This happens because EfficientAD [returns a dictionary](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/models/efficient_ad/torch_model.py#L351). You can modify the code of torch inferencer in the following way to avoid this:
```python
predictions = predictions["anomaly_map_combined"] 
```
before if statements, or change the model to return only `anomaly_map_combined`.

I think this was discussed somewhere before, but I can't find that thread. @alexriedel1 @samet-akcay do you think that this should be changed to just return combined map, since other two are not used anywhere as far as I see?

**Reply by alexriedel1** [reply]
the student and autoencoder maps are used for calculating the quantiles after training 
https://github.com/openvinotoolkit/anomalib/blob/9a33639d4daaf521e15ae6e38d25bc11428a5462/src/anomalib/models/efficient_ad/lightning_model.py#L172-L173

the torch model could be modified to output only the combined map at inference regardless of this, but for a quick fix the inferencer can be modified @wwt123-star

**Reply by blaz-r** [reply]
Thanks for clarifying, I didn't see that it's used there, but it makes sense. I think something will need to change, because right now I think that Efficient AD can't really be used with torch or openvino inferencers as-is.

**Reply by alexriedel1** [reply]
Yes, i think the way will be to add an argument to the torch model forward method that specifies whether the st and ae maps are returned of just the combined map.
